src/gui/connectionvalidator.cpp

```diff
@@ -225,32 +226,57 @@ void ConnectionValidator::slotAuthSuccess()
 void ConnectionValidator::checkServerCapabilities()
 {
     // The main flow now needs the capabilities
-    JsonApiJob *job = new JsonApiJob(_account, QLatin1String("ocs/v1.php/cloud/capabilities"), this);
+    JsonApiJob *job = new JsonApiJob(_account, QStringLiteral("ocs/v2.php/cloud/capabilities"), {}, {}, this);
     job->setTimeout(timeoutToUse);
-    QObject::connect(job, &JsonApiJob::jsonReceived, this, &ConnectionValidator::slotCapabilitiesRecieved);
-    job->start();
-}
 
-void ConnectionValidator::slotCapabilitiesRecieved(const QJsonDocument &json)
-{
-    auto caps = json.object().value("ocs").toObject().value("data").toObject().value("capabilities").toObject();
-    qCInfo(lcConnectionValidator) << "Server capabilities" << caps;
-    _account->setCapabilities(caps.toVariantMap());
+    QObject::connect(job, &JsonApiJob::finishedSignal, this, [job, this] {
+        auto caps = job->data().value("ocs").toObject().value("data").toObject().value("capabilities").toObject();
+        qCInfo(lcConnectionValidator) << "Server capabilities" << caps;
+        _account->setCapabilities(caps.toVariantMap());
 
-    // New servers also report the version in the capabilities
-    QString serverVersion = caps["core"].toObject()["status"].toObject()["version"].toString();
-    if (!serverVersion.isEmpty() && !setAndCheckServerVersion(serverVersion)) {
-        return;
-    }
+        // New servers also report the version in the capabilities
+        QString serverVersion = caps["core"].toObject()["status"].toObject()["version"].toString();
```

Comment:
```suggestion
        auto serverVersion = caps["core"].toObject()["status"].toObject()["version"].toString();
```

Reply:
Nope as a rule of the thumb better not use auto with QString and QStringBuilder enabled.